driver.go

```diff
@@ -60,6 +61,14 @@ func (d MySQLDriver) Open(dsn string) (driver.Conn, error) {
 	mc.parseTime = mc.cfg.ParseTime
 	mc.strict = mc.cfg.Strict
 
+	// Call startWatcher for context support (From Go 1.8)
+	type starter interface {
```

Comment:
I think that scopes are the smaller the better.
`starter` interface is only used in this function, so `starter` interface is in this function.